pysqldb3/geopackage.py

```diff
@@ -374,15 +374,15 @@ def read_gpkg(self, precision=False, private=True, gpkg_encoding=None, print_cmd
             self.gpkg_name = os.path.basename(filename)
             self.path = os.path.dirname(filename)
 
-        if self.table:
-            self.table = self.table.lower()
-        
-        if self.table_exists():
+        if table:
+            table = table.lower()
+
+        if dbo.table_exists(schema = schema, table = table):
             # Clean up spatial index
-            self.del_indexes()
+            self.del_indexes(dbo = dbo, schema = schema, table = table)
 
-            print('Deleting existing table {s}.{t}'.format(s=self.schema, t=self.table))
-            self.dbo.drop_table(schema=self.schema, table=self.table)
+            print('Deleting existing table {s}.{t}'.format(s=schema, t=table))
+            dbo.drop_table(schema=schema, table=table)
 
         path = self.path
         full_path = os.path.join(path, self.gpkg_name)
```

Comment:
for organizational reasons i would say bulk read should be seperate functions that can get called by the read if needed